pkg/controller/database/claim.go

```diff
@@ -199,21 +201,17 @@ func ConfigureMyCloudsqlInstance(_ context.Context, cm resource.Claim, cs resour
 	return nil
 }
 
-func translateVersion(version, versionPrefix string) string {
-	if version == "" {
-		return ""
+func translateVersion(version, versionPrefix string) *string {
+	r := ""
+	if version != "" {
+		r = fmt.Sprintf("%s_%s", versionPrefix, strings.Replace(version, ".", "_", -1))
 	}
-	return fmt.Sprintf("%s_%s", versionPrefix, strings.Replace(version, ".", "_", -1))
+	return &r
 }
 
 func checkEmptySpec(spec *v1alpha2.CloudsqlInstanceSpec) {
-	if spec.Labels == nil {
-		spec.Labels = map[string]string{}
-	}
-	if spec.AuthorizedNetworks == nil {
-		spec.AuthorizedNetworks = []string{}
-	}
-	if spec.StorageGB == 0 {
-		spec.StorageGB = v1alpha2.DefaultStorageGB
+	if gcp.Int64Value(spec.ForProvider.Settings.DataDiskSizeGb) == 0 {
```

Comment:
The comment for this field implies it's optional because it's not required for first generation instances. Do we really want to set a default? It seems like this is potentially second guessing resource class authors. Perhaps we should just let them author an invalid resource class? I'm guessing there's plenty of other ways to do that already.